ui-v2/app/adapters/application.js

```diff
@@ -7,14 +7,16 @@ export const NSPACE_QUERY_PARAM = 'ns';
 export default Adapter.extend({
   repo: service('settings'),
   client: service('client/http'),
-  formatNspace: function(nspace) {
+  formatNspace: function(queryParamKeys) {
     if (config.CONSUL_NSPACES_ENABLED) {
-      return nspace !== '' ? { [NSPACE_QUERY_PARAM]: nspace } : undefined;
```

Comment:
Renamed `nspace` to `queryParamKeys` to better reflect that the argument passed here is an object like
```
{
  NSPACE_QUERY_PARAM: "ns",
  DATACENTER_QUERY_PARAM: "dc"
}
```
for both `formatNspace` and `formatDatacenter` functions. As such, I think the conditional here would never actually be false (I think you'd get a comparison like `{} !== ''` if no query param keys were passed). I'm not sure if this should return an empty object if `config.CONSUL_NSPACES_ENABLED` or if current behavior of undefined is preferable.

Reply:
This was being called wrong somehow, looks correct testing with #6973